frontend/packages/dev-console/src/components/import/section/useSetResourceType.ts

```diff
@@ -0,0 +1,91 @@
+import * as React from 'react';
+import { FlagsObject } from '@console/internal/reducers/features';
+import { FLAG_KNATIVE_SERVING_SERVICE } from '@console/knative-plugin';
+import { useUserSettings, useUserSettingsCompatibility } from '@console/shared';
+import {
+  LAST_RESOURCE_TYPE_STORAGE_KEY,
+  PREFERRED_RESOURCE_TYPE_LOCAL_STORAGE_KEY,
+  PREFERRED_RESOURCE_TYPE_USER_SETTING_KEY,
+} from '../../../const';
+import { Resources } from '../import-types';
+
+const LAST_USED_RESOURCE_KEY = 'latest';
+const fieldName = 'resources';
+
+export const useSetResourceType = (
+  formType: string,
+  flags: FlagsObject,
+  setFieldValue: (field: string, value: any, shouldValidate?: boolean) => void,
+) => {
+  const defaultResourceType = flags[FLAG_KNATIVE_SERVING_SERVICE]
+    ? Resources.KnativeService
+    : Resources.Kubernetes;
+
+  const [
+    preferredResourceType,
+    setPreferredResourceType,
+    preferredResourceTypeLoaded,
+  ] = useUserSettings<string>(PREFERRED_RESOURCE_TYPE_USER_SETTING_KEY, defaultResourceType, true);
+
+  const [resourceType, setResourceType, resourceTypeLoaded] = useUserSettingsCompatibility<string>(
+    LAST_RESOURCE_TYPE_STORAGE_KEY,
+    PREFERRED_RESOURCE_TYPE_LOCAL_STORAGE_KEY,
+    defaultResourceType,
+    true,
+  );
```

Comment:
why we need both `useUserSettingsCompatibility` and `useUserSettings`?

Reply:
Removed `useUserSettingsCompatibility` and used `useUserSettings` instead.